controllers/osartifact_controller.go

```diff
@@ -57,6 +77,12 @@ func genOwner(artifact buildv1alpha1.OSArtifact) []metav1.OwnerReference {
 //+kubebuilder:rbac:groups=build.kairos.io,resources=osartifacts/status,verbs=get;update;patch
 //+kubebuilder:rbac:groups=build.kairos.io,resources=osartifacts/finalizers,verbs=update
 
+// TODO: Is this ^ how I should have created rbac permissions for the controller?
```

Comment:
reminder

Reply:
Although it's possible, afaiu we have the role_custom exactly for this reason, so that we can add permissions without annotations. If that's true, they I'll just remove this TODO. @mudler ?

Reply:
it needs to use the annotation in comments otherwise your changes would get wiped out when running `make manifests`

Reply:
similarly we should also drop permissions to create services and deployments (before the controller could create both, but now just jobs)

Reply:
I run `make bundle` (which runs `make manifests` ) and it didn't drop anything from the `role_custom` file. By adding this annotation:

```
+//+kubebuilder:rbac:groups=,resources=pods/exec,verbs=create
```

I got it do add the permission but it did so in the `config/rbac/role.yaml` file. I guess that's the only file it keeps track of, that's why I assumed we create the `role_custom.yaml` one so that we could completely bypass kubebuilder.

Reply:
The deployments permission was already changed to `jobs`. I pushed a commit to remove the permissions on services. There was no annotations for these. I guess they were added by hand in the `role_custom.yaml`.